src/puppetlabs/trapperkeeper/services/status/status_service.clj

```diff
@@ -0,0 +1,30 @@
+(ns puppetlabs.trapperkeeper.services.status.status-service
+  (:require [clojure.tools.logging :as log]
+            [compojure.core :as compojure]
+            [puppetlabs.trapperkeeper.core :refer [defservice]]
+            [puppetlabs.trapperkeeper.services :refer [service-context]]
+            [puppetlabs.trapperkeeper.services.status.status-core :as core]
+            [puppetlabs.trapperkeeper.services.webrouting.webrouting-service :as webrouting-service]))
+
+(defprotocol StatusService
+  (register-status [this service-name service-version status-version status-fn]
```

Comment:
I agree that `service-version` should be optional, and that it might be useful to have helper functions that can pull out the service's version from the lein metadata.

I'm not so sure about `status-version`. Compared to `service-version`, actually supplying this as an argument seems pretty easy, and it's not going to change as often as `service-version`. If `status-version` is optional but someone registers two different status functions for one service, what happens (this is actually also potentially a current problem - two status functions can be registered without changing the status version... )? Do we just register both of them and choose randomly which to call for the roll-up endpoint? Or do we only register one of them?